flink-tests/src/test/java/org/apache/flink/test/accumulators/AccumulatorLiveITCase.java

```diff
@@ -18,292 +18,188 @@
-		jobID = jobGraph.getJobID();
+		JobGraph jobGraph = env.getStreamGraph().getJobGraph();
 
-		verifyResults();
+		submitJobAndVerifyResults(jobGraph);
 	}
 
-	private static void verifyResults() {
-		new JavaTestKit(system) {{
-
-			ActorGateway selfGateway = new AkkaActorGateway(getRef(), jobManagerGateway.leaderSessionID());
-
-			// register for accumulator changes
-			jobManagerGateway.tell(new TestingJobManagerMessages.NotifyWhenAccumulatorChange(jobID), selfGateway);
-			expectMsgEquals(TIMEOUT, true);
-
-			// submit job
-
-			jobManagerGateway.tell(
-					new JobManagerMessages.SubmitJob(
-							jobGraph,
-							ListeningBehaviour.EXECUTION_RESULT),
-					selfGateway);
-			expectMsgClass(TIMEOUT, JobManagerMessages.JobSubmitSuccess.class);
-
-			TestingJobManagerMessages.UpdatedAccumulators msg = (TestingJobManagerMessages.UpdatedAccumulators) receiveOne(TIMEOUT);
-			Map<String, Accumulator<?, ?>> userAccumulators = msg.userAccumulators();
+	private static void submitJobAndVerifyResults(JobGraph jobGraph) throws Exception {
 
-			ExecutionAttemptID mapperTaskID = null;
-
-			ExecutionAttemptID sinkTaskID = null;
-
-			/* Check for accumulator values */
-			if (checkUserAccumulators(0, userAccumulators)) {
-				LOG.info("Passed initial check for map task.");
-			} else {
-				fail("Wrong accumulator results when map task begins execution.");
-			}
-
-			int expectedAccVal = 0;
-
-			/* for mapper task */
-			for (int i = 1; i <= NUM_ITERATIONS; i++) {
-				expectedAccVal += i;
-
-				// receive message
-				msg = (TestingJobManagerMessages.UpdatedAccumulators) receiveOne(TIMEOUT);
-				userAccumulators = msg.userAccumulators();
-
-				LOG.info("{}", userAccumulators);
-
-				if (checkUserAccumulators(expectedAccVal, userAccumulators)) {
-					LOG.info("Passed round #" + i);
-				} else if (checkUserAccumulators(expectedAccVal, userAccumulators)) {
-					// we determined the wrong task id and need to switch the two here
-					ExecutionAttemptID temp = mapperTaskID;
-					mapperTaskID = sinkTaskID;
-					sinkTaskID = temp;
-					LOG.info("Passed round #" + i);
-				} else {
-					fail("Failed in round #" + i);
-				}
-			}
-
-			msg = (TestingJobManagerMessages.UpdatedAccumulators) receiveOne(TIMEOUT);
-			userAccumulators = msg.userAccumulators();
-
-			if (checkUserAccumulators(expectedAccVal, userAccumulators)) {
-				LOG.info("Passed initial check for sink task.");
-			} else {
-				fail("Wrong accumulator results when sink task begins execution.");
-			}
+		ClusterClient<?> client = MINI_CLUSTER_RESOURCE.getClusterClient();
 
-			/* for sink task */
-			for (int i = 1; i <= NUM_ITERATIONS; i++) {
+		client.setDetached(true);
+		client.submitJob(jobGraph, AccumulatorLiveITCase.class.getClassLoader());
 
-				// receive message
-				msg = (TestingJobManagerMessages.UpdatedAccumulators) receiveOne(TIMEOUT);
+		try {
+			NotifyingMapper.notifyLatch.await();
+			Thread.sleep(HEARTBEAT_INTERVAL * 4); // wait for heartbeat
```

Comment:
Isn't this test-flakiness waiting to happen? We could run this in a loop and wait until the accumulators are available with a `Deadline`. If `getAccumulators()` returned a Future we could even use the utilities in `FutureUtils`, but can't fix that now, unfortunately... 😓

Reply:
Well....yeah we could use a deadline and polling, probably will be even faster.

Reply:
@aljoscha Done; also found a way to make use of `FutureUtils`.